File: pkg/game/actor.go
package game

import "github.com/hueypark/marsettler/core/math/vector"

// Actor is basic object in world.
type Actor struct {
	id       int64
	position vector.Vector
}

// ID returns id.
func (a *Actor) ID() int64 {
	return a.id
}

// Position is actor's position.
func (a *Actor) Position() vector.Vector {
	return a.position
}

File: Server/Source/Engine/World/Actor.cpp
#include "Actor.h"

Actor::Actor(const int64_t id, const Vector& position) : m_id(id), m_position(position)
{
}

Actor::~Actor()
{
}

int64_t Actor::ID() const
{
	return m_id;
}

Vector Actor::Position() const
{
	return m_position;
}
